frontend/components/Cell.js

```diff
@@ -117,16 +120,129 @@ export const Cell = ({
     const class_code_folded = code_folded && cm_forced_focus == null
 
     let show_input = errored || class_code_differs || !class_code_folded
+    const prepareFileBase64 = (file) =>
+        new Promise((resolve, reject) => {
+            const { name, type } = file
+            const fr = new FileReader()
+            fr.onerror = () => reject("Failed to read file!")
+            fr.onloadstart = () => {}
+            fr.onprogress = ({ loaded, total }) => {}
+            fr.onload = () => {}
+            fr.onloadend = ({ target: { result } }) => resolve({ fileBase64: String(result).replace(/.*base64,/, ""), name, type })
+            fr.readAsDataURL(file)
+        })
 
+    const uploadAndCreateCodeTemplate = async (file) => {
+        if (!(file instanceof File)) return " #  File can't be read"
+        setSavingFile(true)
+
+        const {
+            message: { success, file_path, file_name },
+        } = await prepareFileBase64(file).then(
+            (preparedObj) => {
+                return requests.write_file(cell_id, preparedObj)
+            },
+            () => alert("Pluto can't save this file 😥")
+        )
+        setSavingFile(false)
+        if (!success) {
+            alert("Pluto can't save this file 😥")
+            return "# File save failed"
+        }
+
+        const [name, extension] = file_name.split(".").map((a) => a.replace(/["\-,#@!\%\s+\;()$&*\[\]\{\}']/g, ""))
+        switch (file?.type) {
```

Comment:
Relying on MIME types seems to be rather unstable for me, e.g. CSV files have on Windows (at least for me) Excel MIME type, SQLite db files no MIME type.
Probably just relying on the file extension would be more reliable.

Reply:
updated -- moved this code in julia